src/Features/Core/Portable/CodeRefactorings/UseNamedArguments/AbstractUseNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -35,10 +35,12 @@ protected abstract class Analyzer<TBaseArgumentSyntax, TArgumentSyntax, TArgumen
                 }
 
                 var token = root.FindToken(context.Span.Start);
-                if (IsCloseParenOrComma(token))
+                if (token.Span.Start == context.Span.Start &&
+                    IsCloseParenOrComma(token))
                 {
-                    if (token.Span.Start != context.Span.Start ||
-                        IsCloseParenOrComma(token = token.GetPreviousToken()))
+                    token = token.GetPreviousToken();
+                    if (token.Span.End != context.Span.Start || 
+                        IsCloseParenOrComma(token))
```

Comment:
I think the walk up should stop immediately at the first parameterlist.  that would make more sense to me.